test/weights.jl

```diff
@@ -10,6 +10,13 @@ weight_funcs = (weights, aweights, fweights, pweights)
     @test isa(f([1., 2., 3.]), AbstractWeights{Float64})
     @test isa(f([1 2 3; 4 5 6]), AbstractWeights{Int})
 
+    # 371
+    @test isa(f([1, 2, 3])[1:2], AbstractWeights{Int})
```

Comment:
Better make the test more specific and check that the result is of the type of the input.

Reply:
Or rather `typeof(f(x)[1:2]) == typeof(f(x))`. Probably cleaner by iterating other types/constructors rather than functions (e.g. `FrequencyWeights` instead of `fweights`), so that you can do `isa T` instead of `typeof(f(x))`.